pkg/cache/cache.go

```diff
@@ -554,8 +570,21 @@ func (c *Cache) ForgetWorkload(w *kueue.Workload) error {
 	return nil
 }
 
-// Usage reports the used resources and number of workloads admitted by the ClusterQueue.
-func (c *Cache) Usage(cqObj *kueue.ClusterQueue) ([]kueue.FlavorUsage, int, error) {
+// ReservationStats reports the reserved resources and number of workloads holding them in the ClusterQueue.
+func (c *Cache) ReservationStats(cqObj *kueue.ClusterQueue) ([]kueue.FlavorUsage, int, error) {
```

Comment:
We should return all the values (usage, reservations and their counters) under the same lock.
Otherwise they might return state that are incompatible.

I would return a struct that contains the 4 fields, as opposed to returning 4 values, which can be confusing from a readability perspective.